### PR TITLE
Update survey, voter reg pop-up, and post-share affirmation to use the new modal implementation!

### PR DESCRIPTION
**Pull request by DFurnes**
### What does this PR do?

This PR updates the post-share affirmation on the Share Action to use the new & improved `Modal` component. It's a lot cleaner, nice! I've also done a little refactoring in the `ShareAction` to flatten out an unnecessary nested function inside the render method.

### Any background context you want to provide?
Another step towards removing the old modal implementation!

For fdbd433, here's [what it looked like before](https://user-images.githubusercontent.com/583202/38706597-bece35ee-3e7b-11e8-84b5-149f95b28c92.png). No good! 🙅‍♂️

### What are the relevant tickets/cards?
N/A?

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.
* [ ] Added screenshot to PR description of related front-end updates on **small** screens.
* [ ] Added screenshot to PR description of related front-end updates on **medium** screens.
* [ ] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

**Comment by DFurnes** [comment]
Just gotta update tests to look for the new `Modal` component in the mornin'! 🌞

**Comment by weerd** [comment]
@DFurnes while in here, would you want to update the tracking stuff to use the new middleware/helper functions? Or maybe best as a different PR?

**Comment by weerd** [comment]
Actually, @DFurnes might be best for a later PR.... updating the tracking would be a bit involved and outside scope of this Modal updates PR. Nevermind!

File: resources/assets/components/utilities/Modal/modal.scss [hunk 1]
```diff
@@ -32,6 +32,7 @@
 
 .modal {
   margin: auto;
+  min-width: 400px;
```
Comment (weerd) [inline resources/assets/components/utilities/Modal/modal.scss]: Nice! I recently encountered a super-small modal, cause I had forgotten to add a min-width! Thanks 💃

**Comment by DFurnes** [comment]
@weerd @mendelB Updated the tests in 3021631 and ready for final review!

**Comment by DFurnes** [comment]
Surprise! Updated a few more modals to use the new implementation:

🚫 Removed `BLOCK_MODAL`, which was used for the old `/blocks/:id` route.

🚀 Refactored the `ModalLauncher` to use new modal implementation, and a [render prop](https://reactjs.org/docs/render-props.html) (passed in from wherever it's rendered, so it's nice and easy to add more "launchers" in the future).

🗑 Removed `VOTER_REGISTRATION_MODAL` and `SURVEY_MODAL` from the old modal switch (keeping the components in the same place for now, but we should eventually move).

**Comment by DFurnes** [comment]
Also removed the old `REPORTBACK_UPLOADER_MODAL` and `ReportbackUploaderModal`, which was from an unfinished prototype that was removed in #832 when we re-built `/modal/:id` routes.

**Comment by DFurnes** [comment]
Working on refactoring the post-reportback modal next, but will push that up in a separate PR! 😇

File: resources/assets/components/ModalLauncher/ModalLauncher.test.js [hunk 2]
```diff
@@ -51,71 +43,66 @@ describe('The ModalLauncher component', () => {
...
 
   it('opens the correct modal type', () => {
-    const openModalMock = mountModal();
-
+    const wrapper = getShallow();
     jest.runAllTimers();
 
-    expect(openModalMock.mock.calls[0][0]).toBe(MODAL_TYPE);
+    expect(wrapper.update().find('Modal')).toHaveLength(1);
+    // expect(openModalMock.mock.calls[0][0]).toBe(MODAL_TYPE);
```
Comment (mendelB) [inline resources/assets/components/ModalLauncher/ModalLauncher.test.js]: [A reminder of darker times](http://halachipedia.com/index.php?title=Zecher_LeChurban)? 😂

File: resources/assets/components/ModalLauncher/ModalLauncher.js [hunk 3]
```diff
@@ -48,19 +48,22 @@ class ModalLauncher extends React.Component {
...
       env(`${type.toUpperCase()}_ENABLED`) && !shouldNotSee && !isDismissed
     );
-  }
+  };
 
   render() {
-    return null;
+    return this.state.showModal ? (
+      <Modal onClose={() => this.setState({ showModal: false })}>
+        {this.props.render()}
```
Comment (mendelB) [inline resources/assets/components/ModalLauncher/ModalLauncher.js]: 🐱 🐭

File: resources/assets/components/Campaign/Campaign.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ const Campaign = props => (
       <ModalLauncherContainer
         type="nps_survey"
         countdown={60}
-        modalType={SURVEY_MODAL}
+        render={() => <SurveyModalContainer />}
```
Comment (weerd) [inline resources/assets/components/Campaign/Campaign.js]: Cooool!

File: resources/assets/components/ModalLauncher/ModalLauncher.test.js (hunk 2 above)
Comment (DFurnes) [reply resources/assets/components/ModalLauncher/ModalLauncher.test.js]: Oooh I forgot to update that one! I'll sneak it in my next PR! 🔜

**Pull request merged**